### DISCUSSION TITLE
Appium Flutter Driver Can not  take screenshoot if error or timeout

### DISCUSSION THREAD

**Discussion by freak-fahmi** [webdriverio/webdriverio#9053] [Community]
Hallo All

I use appium flutter driver in webdriverIO, framework cucumber
I setup After Step 

afterStep: async function (step, scenario, { error, duration, passed }, context) {
    // if (error) {
        await browser.saveScreenshoot();
    //    }
    },

**Comment by christian-bromann** [comment]
@freak-fahmi you will have to be much more specific about the error message and stack for everyone to help you out here. I can't guess what is going on on your machine.

**Reply by freak-fahmi** [reply]
@christian-bromann  thankyou for respons

wdio.conf.js
const path = require('path');
const { generate } = require('multiple-cucumber-html-reporter');
const cucumberJson = require("wdio-cucumberjs-json-reporter").default;
const allureReporter = require('@wdio/allure-reporter').default
const allure = require('allure-commandline')
// require('dotenv').config();
// var browserstack = require('browserstack-local');

exports.config = {
    runner: 'local',
    maxInstances: 1, 
    logLevel: 'debug',
    bail: 0,
    baseUrl: '',
    waitforTimeout: 10000,
    connectionRetryTimeout: 120000,
    connectionRetryCount: 3,
    services: [
        'appium'
    ],
    appium: {
        command: 'appium',
        args: {},
    },
    
    framework: 'cucumber',
    cucumberOpts: {
        backtrace: false,
        requireModule: [],
        failAmbiguousDefinitions: false,
        failFast: false,
        ignoreUndefinedDefinitions: false,
        name: [],
        profile: [],
        // require: [
        //     './src/stepDefinitions/*.steps.js'
        // ],
        snippetSyntax: undefined,
        snippets: true,
        source: true,
        strict: false,
        tagsInTitle: false,
        timeout: 30000,
        retry: 0
    },
    
    reporters: [['allure', {
        outputDir: 'allure-results',
        disableWebdriverStepsReporting: true,
        disableWebdriverScreenshotsReporting: false,
    }]],
    
    mochaOpts: {
        ui: 'bdd',
        timeout: 60000
    },
    
    afterStep: async function (step, scenario, { error, duration, passed }, context) {
    // if (error) {
    //   browser.takeScreenshot();
      await driver.takeScreenshot();
    // }
    },

     onComplete: function() {
        const reportError = new Error('Could not generate Allure report')
        const generation = allure(['generate', 'allure-results', '--clean'])
        return new Promise((resolve, reject) => {
            const generationTimeout = setTimeout(
                () => reject(reportError),
                5000)

            generation.on('exit', function(exitCode) {
                clearTimeout(generationTimeout)

                if (exitCode !== 0) {
                    return reject(reportError)
                }

                console.log('Allure report successfully generated')
                resolve()
            })
        })
    }

}

android.conf.js
const { config } = require('./wdio.conf');
const AndroidInfo = require('./android.info');
const path = require('path');

// Appium capabilities
config.capabilities = [
    {
        path: '/wd/hub',
        platformName: 'Android',
        noReset: true,
        fullReset: false,
        autoGrantPermissions: true,
//        autoAcceptAlerts: true,
//        maxInstances: 1,
        automationName: 'flutter',
        deviceName: AndroidInfo.deviceName1(),
        platformVersion: AndroidInfo.platFormVersion1(),
        app: path.resolve(`./apps/${AndroidInfo.appName()}`),
        
        // systemPort: 8210
    }

];

// config.cucumberOpts.tagExpression = '@androidApp'; // pass tag to run tests specific to android

config.port = 4723;

config.services = [
    'appium'
];

config.appium = {
    command: 'appium',
    args: {},
};

config.path = '/wd/hub';

// config.specs = [
//     './src/mobile/featureFiles/*.feature'
// ];

config.specs = [
    [
        './features/**/001_login.feature',
    ]
    // ,
    // article: [
    //    './src/mobile/featureFiles/02.article.feature'
    // ]
];

config.cucumberOpts = {
    
    require: [
        './features/step-definitions/*.js'
    ],
};

exports.config = config;

.fetaure
 Feature: Login AGEN46

     Scenario Outline: As a user, I can log into the AGEN46

         Given I am on the login page
         When I lewati splash screen
         When I login for the first time with <username> and <password>
         When I click button masuk
         Then I should see a element present

         Examples:
             | username    | password  |
             | BNIAG100183 | Ipybni06! |

steps.js
const { Given, When, Then } = require('@wdio/cucumber-framework');
const LoginPageTest = require('../pageobjects/login_page');


//Given(/^I am on the login page$/, async () => {
//    console.log('open aplikasi')
//});

Given(/^I am on the login page$/, {timeout: 20000}, async() => {
    console.log ('Start Aplication')
    await LoginPageTest.LaunchApp()
    // await browser.pause(10000);
});

When(/^I lewati splash screen$/, async() => {
    await LoginPageTest.lewatiSplahScreen()

});

When(/^I login for the first time with (\w+) and (.+)$/, async (username, password) => {
    await LoginPageTest.loginFirstInstall(username, password)
});

When(/^I click button masuk$/, async() => {
    await LoginPageTest.clickButtonMasuk()

});


When(/^I login with (\w+) and (.+)$/, async (username, password) => {
    await LoginPageTest.login(username, password)
});


Then(/^I should see a element present$/, async () => {
    await browser.pause(20000);
    await LoginPageTest.VerifyHomeDashboard()
    // await expect(SecurePage.flashAlert).toHaveTextContaining(message);
});

page.js
const find = require('appium-flutter-finder');
var sleep = require('system-sleep');
const assert = require('assert');
const { default: ChromeDriverService } = require('wdio-chromedriver-service');


class LoginPageTest {
    static btnLewatiSplashScreen = find.byValueKey('btnLewatisplashscreen');
    static btnDownloadSplashscreen = find.byValueKey('btnDownloadsplashscreen');
    static btnLewati = find.byValueKey('buttonLewati');
    static btnMasuk = find.byValueKey('buttonMasukOnBoarding');
    static btnMasukLogin = find.byValueKey('btnMasukLogin');
    static txtUsername = find.byValueKey('usernameTextField');
    static txtPassword = find.byValueKey('passwordTextField');
    static btnOK = find.byValueKey('buttonOKallert');
    static btnLanjutHomedashboard = find.byValueKey('buttonLanjutDashboard');
    static btnLanjut2 = find.byValueKey('btnLanjut2');
    static btnMulai = find.byValueKey('btnMulai');
    static iconUbahMenuFav = find.byValueKey('ubahTextField');


    async LaunchApp(){
        await assert.strictEqual(await driver.execute('flutter:checkHealth'), 'ok');
        await driver.execute('flutter:clearTimeline');
        await driver.execute('flutter:forceGC');
        await driver.execute('flutter:waitFor',(LoginPageTest.btnLewatiSplashScreen));
        await expect (LoginPageTest.btnLewatiSplashScreen).toBeDisplayed
        console.log ('Open Aplikasi');
    }

    async lewatiSplahScreen() {
        // await driver.execute('flutter:setFrameSync', false, 3500);
        await driver.elementClick(LoginPageTest.btnLewatiSplashScreen);
        await driver.elementClick(LoginPageTest.btnLewati);
        await driver.elementClick(LoginPageTest.btnMasuk);
    }
    async loginFirstInstall(username, password) {
        await driver.execute('flutter:setFrameSync', false, 3500);
        await driver.elementClick(LoginPageTest.btnOK);
        await driver.elementSendKeys(LoginPageTest.txtUsername, username)
        await driver.elementSendKeys(LoginPageTest.txtPassword, password)
    }

    async clickButtonMasuk(){
        await driver.elementClick(LoginPageTest.btnMasukLogin);
        await driver.elementClick(LoginPageTest.btnLewatiSplashScreen);
        await driver.elementClick(LoginPageTest.btnLanjutHomedashboard);
        await driver.elementClick(LoginPageTest.btnLanjut2);
        await driver.elementClick(LoginPageTest.btnMulai);
    }

    async login(username, password) {
        await driver.execute('flutter:setFrameSync', false, 3500);
        await driver.elementClick(LoginPageTest.btnLewatiSplashScreen);
        await driver.elementClick(LoginPageTest.btnOK);
        await driver.elementSendKeys(LoginPageTest.txtUsername, username)
        await driver.elementSendKeys(LoginPageTest.txtPassword, password)
        await driver.elementClick(LoginPageTest.btnMasukLogin);
        await driver.elementClick(LoginPageTest.btnLewatiSplashScreen);
    }

    async VerifyHomeDashboard (){
            // await driver.execute('flutter:waitForAbsent', LoginPageTest.btnLanjutHomeDashboard)
            // await assert.strictEqual((LoginPageTest.btnLanjutHomeDashboard), 'buttonLanjutDashboard');
            await expect (LoginPageTest.btnLanjutHomeDashboard).toBeDisplayed
        }
}

module.exports = new LoginPageTest();

logfile
[logfile.txt](https://github.com/webdriverio/webdriverio/files/9835558/logfile.txt)

screenshoot:
![image](https://user-images.githubusercontent.com/73210813/197108050-e75117fb-b2cd-4e4a-95ab-fd06902ef69e.png)

every timeot can't screenshot
please help, thankyou
